test/e2e/framework/pods.go

```diff
@@ -109,3 +112,54 @@ func (c *PodClient) mungeSpec(pod *api.Pod) {
 }
 
 // TODO(random-liu): Move pod wait function into this file
+
+func CreateMemhogPod(f *Framework, genName string, ctnName string, res api.ResourceRequirements) *api.Pod {
+	env := []api.EnvVar{
+		{
+			Name: "MEMORY_LIMIT",
+			ValueFrom: &api.EnvVarSource{
+				ResourceFieldRef: &api.ResourceFieldSelector{
+					Resource: "limits.memory",
+				},
+			},
+		},
+	}
+
+	// If there is a limit specified, pass 80% of it for -mem-total, otherwise use the downward API
+	// to pass limits.memory, which will be the total memory available.
+	// This helps prevent a guaranteed pod from triggering an OOM kill due to it's low memory limit,
+	// which will cause the test to fail inappropriately.
+	var memLimit string
+	if limit, ok := res.Limits["memory"]; ok {
+		memLimit = strconv.Itoa(int(
+			float64(limit.Value()) * 0.8))
+	} else {
+		memLimit = "$(MEMORY_LIMIT)"
+	}
+
+	pod := &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			GenerateName: genName,
+		},
+		Spec: api.PodSpec{
+			RestartPolicy: api.RestartPolicyNever,
+			Containers: []api.Container{
+				{
+					Name:            ctnName,
+					Image:           "gcr.io/google-containers/stress:v1",
+					ImagePullPolicy: "Always",
+					Env:             env,
+					// 60 min timeout * 60s / tick per 10s = 360 ticks before timeout => ~11.11Mi/tick
+					// to fill ~4Gi of memory, so initial ballpark 12Mi/tick.
+					// We might see flakes due to timeout if the total memory on the nodes increases.
+					Args:      []string{"-mem-alloc-size", "12Mi", "-mem-alloc-sleep", "10s", "-mem-total", memLimit},
```

Comment:
Thanks, that works. You are right, actual pod phase is Failed, but the reason is different

```
  "status": {
    "phase": "Failed",
    "message": "Pod Predicate PodFitsHostPorts failed",
    "reason": "PodFitsHostPorts",
    "startTime": "2016-09-07T07:52:08Z"
  }
```